src/corelibs/U2View/src/ov_msa/mca_reads/McaAlternativeMutationsWidget.cpp

```diff
@@ -0,0 +1,162 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#include "McaAlternativeMutationsWidget.h"
+
+#include "../McaEditorSequenceArea.h"
+#include "../McaEditorStatusBar.h"
+
+#include <U2Core/McaDbiUtils.h>
+#include <U2Core/MsaDbiUtils.h>
+#include <U2Core/MultipleChromatogramAlignmentObject.h>
+#include <U2Core/U2AttributeUtils.h>
+#include <U2Core/U2Mod.h>
+#include <U2Core/U2SafePoints.h>
+#include <U2Core/U2OpStatusUtils.h>
+
+namespace U2 {
+
+const QString McaAlternativeMutationsWidget::ALTERNATIVE_MUTATIONS_CHECKED = "ALTERNATIVE_MUTATIONS_CHECKED";
+const QString McaAlternativeMutationsWidget::ALTERNATIVE_MUTATIONS_THRESHOLD = "ALTERNATIVE_MUTATIONS_THRESHOLD";
+
+McaAlternativeMutationsWidget::McaAlternativeMutationsWidget(QWidget* parent)
+    : QWidget(parent) {
+    setupUi(this);
+
+}
+
+void McaAlternativeMutationsWidget::init(MultipleAlignmentObject* _maObject,
+                                         MaEditorSequenceArea* _seqArea,
+                                         MaEditorStatusBar* _statusBar) {
+    SAFE_POINT(_seqArea != nullptr, "MaConsensusModeWidget can not be initialized: MaEditorSequenceArea is nullptr", );
+    SAFE_POINT(_maObject != nullptr, "MaConsensusModeWidget can not be initialized: MultipleAlignmentObject is nullptr", );
+    SAFE_POINT(_statusBar != nullptr, "MaConsensusModeWidget can not be initialized: MaEditorStatusBar is nullptr", );
+
+    seqArea = qobject_cast<McaEditorSequenceArea*>(_seqArea);
+    SAFE_POINT(seqArea != nullptr, "MaConsensusModeWidget can not be initialized: McaEditorSequenceArea is nullptr", );
+
+    mcaObject = qobject_cast<MultipleChromatogramAlignmentObject*>(_maObject);
+    SAFE_POINT(mcaObject != nullptr, "MaConsensusModeWidget can not be initialized: MultipleChromatogramAlignmentObject is nullptr", );
+
+    statusBar = qobject_cast<McaEditorStatusBar*>(_statusBar);
+    SAFE_POINT(mcaObject != nullptr, "MaConsensusModeWidget can not be initialized: McaEditorStatusBar is nullptr", );
+
+    mutationsGroupBox->setChecked(false);
+    mutationsThresholdSlider->setValue(99);
+
+    connect(mutationsGroupBox, SIGNAL(toggled(bool)), this, SLOT(sl_updateAlternativeMutations()));
+    connect(updateMutationsPushButton, SIGNAL(pressed()), this, SLOT(sl_updateAlternativeMutations()));
+}
+
+void McaAlternativeMutationsWidget::sl_updateAlternativeMutations() {
+    U2OpStatus2Log os;
+    U2UseCommonUserModStep userModStep(mcaObject->getEntityRef(), os);
+    Q_UNUSED(userModStep);
+
+    mcaObject->updateAlternativeMutations(mutationsGroupBox->isChecked(), mutationsThresholdSlider->value(), os);
+    CHECK_OP(os, );
+
+    updateDb(os);
+    CHECK_OP(os, );
```

Comment:
not: this check is not needed

Reply:
Why? `updateDb` has `os` as argument and it's possible that it (`os`) may be changed.